packages/spatial/src/common/debug/MeshHelperComponent.ts

```diff
@@ -0,0 +1,86 @@
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Ethereal Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import { useDidMount } from '@etherealengine/common/src/utils/useDidMount'
+import { Entity, defineComponent, setComponent, useComponent, useEntityContext } from '@etherealengine/ecs'
+import { NO_PROXY } from '@etherealengine/hyperflux'
+import { matchesGeometry, matchesMaterial } from '@etherealengine/spatial/src/common/functions/MatchesUtils'
+import { useEffect } from 'react'
+import { BufferGeometry, Material, MeshBasicMaterial } from 'three'
+import { useMeshComponent } from '../../renderer/components/MeshComponent'
+import { ObjectLayerMaskComponent } from '../../renderer/components/ObjectLayerComponent'
+import { ObjectLayerMasks } from '../../renderer/constants/ObjectLayers'
+import { useHelperEntity } from './DebugComponentUtils'
+
+export const MeshHelperComponent = defineComponent({
+  name: 'MeshHelperComponent',
+
+  onInit: (entity) => {
+    return {
+      name: 'mesh-helper',
+      geometry: null! as BufferGeometry,
+      material: new MeshBasicMaterial() as Material,
+      layerMask: ObjectLayerMasks.NodeHelper,
+      entity: undefined as undefined | Entity
+    }
+  },
+
+  onSet: (entity, component, json) => {
+    if (!json) return
+    if (typeof json.name === 'string') component.name.set(json.name)
+
+    if (matchesGeometry.test(json.geometry)) component.geometry.set(json.geometry)
+    else throw new Error('MeshHelperComponent: Geometry required for MeshHelperComponent')
+    if (matchesMaterial.test(json.material)) component.material.set(json.material)
+    if (typeof json.layerMask === 'number') component.layerMask.set(json.layerMask)
+  },
+
+  reactor: function () {
+    const entity = useEntityContext()
+    const component = useComponent(entity, MeshHelperComponent)
+    const helperEntity = useHelperEntity(entity, component)
+    const [mesh, geometryState, materialState] = useMeshComponent(
+      helperEntity,
+      component.geometry.value,
+      component.material.value
+    )
+
+    useEffect(() => {
+      setComponent(helperEntity, ObjectLayerMaskComponent, component.layerMask.value)
+    }, [component.layerMask])
+
+    useDidMount(() => {
```

Comment:
what exactly is this useDidMount here distinct from a useEffect, and can these hooks be done inside useMeshComponent since they are using state passed in and mutating state coming from the hook?

Reply:
still would like clarity on this

Reply:
In this case the useDidMount isn't needed, I switched it back to useEffect. MeshHelperComponent is a debug only component, I've renamed to to DebugMeshComponent to make it more explicit. it's a one off pass through component to keep the code concise, I'd rather change this component to not use useMeshComponent than change how useMeshComponent works for everything else